tool/tsh/common/db.go

```diff
@@ -863,6 +863,23 @@ func newDatabaseInfo(cf *CLIConf, tc *client.TeleportClient, route *tlsca.RouteT
 // checkAndSetPrincipalDefaults checks the db route (schema) name and username,
 // and sets them to defaults if necessary.
 func (d *databaseInfo) checkAndSetPrincipalDefaults(cf *CLIConf, tc *client.TeleportClient, db types.Database) error {
+	profile, err := tc.ProfileStatus()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// if either user or db name isn't given as a cli flag, try to populate
+	// user/db name from an active db cert.
+	if cf.DatabaseUser == "" || cf.DatabaseName == "" {
+		routes, err := profile.DatabasesForCluster(tc.SiteName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if route, ok := findActiveDatabase(d.ServiceName, routes); ok {
+			d.Username = route.Username
+			d.Database = route.Database
```

Comment:
should we do this instead of overwriting both?
https://github.com/gravitational/teleport/blob/b5ad2966f8a4f08367ea29f0f50cdfa5edcb6103/tool/tsh/db.go#L1050-L1058